content/en/apps/tutorials/application-tests.md

```diff
@@ -0,0 +1,600 @@
+        cough_duration: '7',
+      }
+    });
+  });
+{{< /highlight >}}
+
+In [line 3](#assessment-form-test-3) above, the method [harness.fillForm()](https://docs.communityhealthtoolkit.org/cht-conf-test-harness/Harness.html#fillForm) fills the specified form with the given input (answers).
+
+Depending on the form design, the number of inputs to be filled can be large. The inputs are often repeated within a single test or across multiple tests with little or no variation. It is a good idea to keep them in a separate file and refer them from the tests as required. You can also introduce some variations in the inputs using function parameters.
+
+Example: `form-inputs.js`
+```js {linenos=table}
+module.exports = {
+  assessments: {
+    no_cough: [
+      ['no']
+    ],
+    cough: (days) => [
+      ['yes'],
+      [ days ]
+    ]
+  }
+};
+```
+
+In the test below, the inputs are used from the `form-inputs.js` file above:
+```js
+const { assessments } = require('../form-inputs');
+...
+const result = await harness.fillForm('assessment', ...assessments.cough('3'));
+```
+
+The test files are usually grouped in folders to read and run them easily. One way of grouping them is by creating folders for each of the components that can be tested: forms, contact summary, tasks, and targets.
+
+![alt_text](tests-dir.png "Test directory structure")
+
+### Testing Forms
+<table>
+  <tr>
+    <th colspan='2'> What do you test? </th>
+  </tr>
+  <tr>
+   <td>Minimum:</td>
+   <td>Filling a form with the most common options should not result in any error.</td>
+  </tr>
+  <tr>
+   <td>Ideal:</td>
+   <td>All input combinations and constraints are tested with the report fields.</td>
+  </tr>
+</table>
+
+[Previous example](#assessment-form-test) demonstrates test for the app form. More test cases can be added by changing the inputs.
+
+You can also test contact forms using test harness. To fill a contact form, use <code>[fillContactForm(contactType, …answers)](https://docs.communityhealthtoolkit.org/cht-conf-test-harness/Harness.html#fillContactForm)</code>.
+
+Example:
+```js {linenos=table}
+const result = await harness.fillContactForm(
+  'district_hospital',
+  ['new_person', 'William Mars', '1990-08-06', '+1234567891', 'female'],
+  ['yes']
+);
+expect(result.errors).to.be.empty;
+expect(result.contacts).excluding(['_id', 'meta']).to.deep.eq([
+  {
+    ...
+  }
+]);
+```
+
+When filling a form, you can also test the field constraints. The example below asserts that a form does not accept birth date in the future:
+```js {linenos=table}
+it(`Throws validation error when birth date is in future`, async () => {
+  const result = await harness.fillForm('delivery', ['yes', '2090-01-02']);
+  expect(result.errors.length).to.equal(1);
+  expect(result.errors[0]).to.include({
+    msg: 'Date cannot be in the future!'
+  });
+});
+```
+
+
+> Note: If a form triggers a task, some use cases of the form can be tested when testing the task later.
```

Comment:
Please replace with {{% alert title="Note" %}} If a form triggers a task, some use cases of the form can be tested when testing the task later.{{% /alert %}}
Reference: https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#notes-and-tips